strawberryfields/backends/states.py

```diff
@@ -1241,15 +1282,17 @@ def poly_quad_expectation(self, A, d=None, k=0, phi=0, **kwargs):
     def number_expectation(self, modes):
         if len(modes) != len(set(modes)):
             raise ValueError("There can be no duplicates in the modes specified.")
+
         mu = self._mu
         cov = self._cov
+
         if len(modes) == 1:
-            return photon_number_mean(mu, cov, modes[0], hbar=self._hbar)
+            return twq.photon_number_mean(mu, cov, modes[0], hbar=self._hbar)
 
         if len(modes) == 2:
```

Comment:
There is now functionality in the Walrus to this for an arbitrary number of modes. I thought @antalszava had added the necessary piping between SF and TW for doing this.

Reply:
that's part of a change I have locally, a missing piece for that too was having the change of TW Fock gate compatibility (the changes from the branch this PR branches off from)